src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
+    (if (true? (:allow_unauthenticated config-map))
+      (assoc rule :allow_unauthenticated true)
```

Comment:
This feels a bit gross and smells bad to me.  Suggestions welcome to improve the flow here.

Maybe get rid of the conditional with something like?

```clj
(->>
  (true? (:allow_unauthenticated config-map)
  (assoc rule :allow_unauthenticated))
```